src/home/substatus.asm

```diff
@@ -471,20 +471,23 @@ NoDamageOrEffectTextIDTable::
 ; return carry if turn holder has Omanyte and its Clairvoyance Pkmn Power is active
 IsClairvoyanceActive::
 	ld a, MUK
-	call CountPokemonIDInBothPlayAreas
+	call CountPokemonWithActivePkmnPowerInBothPlayAreas
 	ccf
 	ret nc
 	ld a, OMANYTE
-	call CountPokemonIDInPlayArea
+	call CountTurnDuelistPokemonWithActivePkmnPower
 	ret
 
 ; returns carry if turn holder's arena card is paralyzed, asleep, confused,
-; and/or toxic gas in play, meaning that attack and/or pkmn power cannot be used
-CheckCannotUseDueToStatus::
-	xor a
+; and/or toxic gas in play (i.e. its pkmn power cannot be used)
+CheckIsIncapableOfUsingPkmnPower_ArenaCard::
+	xor a ; PLAY_AREA_ARENA
 
-; same as above, but if a is non-0, only toxic gas is checked
-CheckCannotUseDueToStatus_OnlyToxicGasIfANon0::
+; returns carry if Pokemon in turn holder's Play Area location in register a
+; cannot use its Pkmn Power
+; input:
+;	a = play area location offset of the Pokémon to check (PLAY_AREA_* constant)
```

Comment:
This description is not correct, if register `a` is non-zero, then status is ignored and only Toxic Gas is considered. It does not have anything to do with Play Area locations. But the current label is also incomplete since it checks for both status and Toxic Gas. Maybe `CheckIsIncapableOfUsingPkmnPower`? And the routine above may be `CheckIsIncapableOfUsingPkmnPower_IgnoreStatus`?

Reply:
I'm forced to disagree. In almost every instance, a play area location offset gets loaded into the a register before this function is called, even if the specifics of this function only care about whether or not the location is the Arena (because an Active Pokémon's Pokémon Power cannot be used if it is Asleep, Confused, or Paralyzed). I think the labels from poketcg2 are about as accurate as you can get. Plus, your `CheckIsIncapableOfUsingPkmnPower_IgnoreStatus` label would imply the opposite of what that first function does, which is to set a to 0 (PLAY_AREA_ARENA) so that the Active Pokémon's status gets checked. The only instance of the game intentionally ignoring the status check is when a Pokémon with a Pokémon Power is put onto the Bench; in that singular instance, $01 (PLAY_AREA_BENCH1) is loaded into a before this function is called.

Reply:
Ah well spotted, now that I look at it it does make sense